src/views/ApplicationPage/index.js

```diff
@@ -36,6 +37,16 @@ export default class application extends Component {
       link_to_resume: undefined,
       why_goldenHack: undefined,
     };
+
+    var jwt = getJwt();
+
+    if (jwt != null) {
+      this.state.email = getEmailFromJwt();
+    } else {
+      this.props.history.push({
+        pathname: "/login",
+      });
```

Comment:
What does this do exactly? Does this put them back to the login page?

Reply:
Yeah it just redirects them to the login page if they haven't yet logged in!